src/entrypoints/accounts-actions/accountRemoval.ts

```diff
@@ -0,0 +1,127 @@
+  /**
+   * The address of the account
+   */
+  account: string;
+  /*
+   * ID associated with the current network.
+   */
+  chainId: number;
+  /*
+   * An optional bytes32 string that will be used for signature replay protection
+   * (Should be omitted, and in that case, a random salt will be generated)
+   */
+  salt?: string;
+};
+
+type DispatchParameters = {
+  /**
+   * The address of the account
+   */
+  account: string;
+};
+
+type RemovalConfig = {
+  previousLastAccountOwner: string;
+  accountToRemove: string;
+};
+
+/**
+ * This function generates a payload to initiate an removal of existing account owner in
+ * Delay Mod's queue. The populated transaction is relay ready, and does not
+ * require additional signing.
+ *
+ * @param parameters - {@link EnqueueParameters}
+ * @param newOwner - {@link `0x${string}`}
+ * @param sign - {@link SignTypedDataCallback}
+ * @returns The signed transaction payload {@link TransactionRequest}
+ *
+ * @example
+ * import { populateRemoveOwnerEnqueue } from "@gnosispay/account-kit";
+ *
+ * const owner: Signer = {};
+ * const enqueueTx = await populateRemoveOwnerEnqueue(
+ *  { account: `0x<address>`, chainId: `<number>` },
+ *  newOwner: `0x${string}`,
+ *  // callback that wraps an eip-712 signature
+ *  ({ domain, primaryType, types, message }) =>
+ *    owner.signTypedData(domain, primaryType, types, message)
+ * );
+ * await relayer.sendTransaction(enqueueTx);
+ */
+export async function populateRemoveOwnerEnqueue(
+  { account, chainId, salt }: EnqueueParameters,
+  removalConfig: RemovalConfig,
+  sign: SignTypedDataCallback
+): Promise<TransactionRequest> {
+  account = getAddress(account);
+  salt = salt || saltFromTimestamp();
+
+  return populateExecuteEnqueue(
+    { account, chainId, salt },
+    createInnerTransaction(account, removalConfig),
+    sign
+  );
+}
+
+/**
+ * Generates a payload that executes an allowance change previously posted to
+ * the Delay Mod. Only works after cooldown seconds and have passed, and before
+ * expiration. The populated transaction is relay ready, and does not require
+ * additional signing.
+ *
+ * @param parameters - {@link DispatchParameters}
+ * @param transaction - {@link AllowanceConfig}
+ * @returns The signed transaction payload {@link TransactionRequest}
+ *
+ * @example
+ * import { populateRemoveOwnerDispatch } from "@gnosispay/account-kit";
+ *
+ * const dispatchTx = await populateRemoveOwnerDispatch(
+ *  { account: `0x<address>` },
+ *  newOwner: `0x${string}`,
+ * );
+ * await relayer.sendTransaction(enqueueTx);
+ */
+export function populateRemoveOwnerDispatch(
+  { account }: DispatchParameters,
+  removalConfig: RemovalConfig
+): TransactionRequest {
+  account = getAddress(account);
+
+  return populateExecuteDispatch(
+    { account },
+    createInnerTransaction(account, removalConfig)
+  );
+}
+
+export function createInnerTransaction(
+  account: string,
+  { previousLastAccountOwner, accountToRemove }: RemovalConfig
```

Comment:
I would call this prevOwner and owner

Reply:
`owner` is kind of unexplicit I'd say, `ownerToRemove` maybe?

Reply:
or prev / owner. I guess additional explanations should come on the code block that precedes the functions. Keeping the var names lean is easier on the reading eye